components/FieldInputWithLabel.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { Text, View } from '../components/Themed';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { TextInput, ScrollView} from 'react-native-gesture-handler';
+import { formatDate } from '../utility/utility';
+import { StyleSheet, KeyboardAvoidingView, Platform, Route } from 'react-native';
+
+
+interface FieldInputWithLabelProps{
+    currentObject: any,
+    property: string,
+    label: string,
+    editable: boolean,
+    valueType?: string,
+    onChangeCallback: (property: string, value: any) => void
+}
+
+export default function FieldInputWithLabel(props : FieldInputWithLabelProps){
```

Comment:
@kendricchung I would argue it's not that specific of a component. It just takes in an object of any kind, and allow you to display a property of that object inside the text field. Granted right now it does look at one specific property ("paymentDate") - but I can change this condition to instead check for a "date" type property, which just means that the current property is a date.

For the 2nd point - I kept it like this because label and input are already broken down enough. "Label" is just a <Text> component, and "Input" are a pre-existing components (DateTimePicker, TextInput) from external libraries.

Reply:
I see but I still think that you should create multiple components instead of relying on property. There are too many dependencies on property mapping to a current value. That is, there are too many things that we need to handle in one component but where as if you create 3 other ones, the logic can be added to that one. We shouldn't try to make a component do too may different things. We want to minimize logic as much as possible such that there are very little complications. Of course there are exceptions but in this case, I think it should be multiple components instead of one.